drivers/scheduler/k8s/k8s.go

```diff
@@ -3153,16 +3153,16 @@ func (k *K8s) Destroy(ctx *scheduler.Context, opts map[string]bool) error {
 	var podList []corev1.Pod
 
 	allRules, err := k8sAutopilot.ListAutopilotRules()
-	if err != nil {
-		return err
-	}
-
-	for _, rule := range allRules.Items {
-		log.InfoD("Cleaning up autopilot rule: %s", rule.Name)
-		err := k8sAutopilot.DeleteAutopilotRule(rule.Name)
-		if err != nil {
-			return err
+	if err == nil {
+		for _, rule := range allRules.Items {
```

Comment:
Error could be nil, but rules could also be empty, I would check if its empty before going through loop and maybe even say no rules were found if it is empty